DLCS.Model/Assets/InfoJsonBuilder.cs

```diff
@@ -48,31 +46,26 @@ public static string GetImageApi2_1Level0(string serviceEndpoint, List<int[]> si
         /// <param name="width">Width of image</param>
         /// <param name="height">Height of image</param>
         /// <returns>info.json string</returns>
-        public static string GetImageApi2_1Level1(string serviceEndpoint, int width, int height, List<int[]> sizes)
+        public static ImageService2 GetImageApi2_1Level1(string serviceEndpoint, int width, int height,
+            List<int[]> sizes)
         {
-            const string template = @"{
-""@context"":""http://iiif.io/api/image/2/context.json"",
-""@id"":""$id$"",
-""protocol"": ""http://iiif.io/api/image"",
-""profile"": [
-  ""http://iiif.io/api/image/2/level1.json"",
-  {
-    ""formats"": [ ""jpg"" ],
-    ""qualities"": [ ""native"",""color"",""gray"" ],
-    ""supports"": [ ""regionByPct"",""sizeByForcedWh"",""sizeByWh"",""sizeAboveFull"",""rotationBy90s"",""mirroring"",""gray"" ]
-  }
-  ],
-  ""width"": $width$,
-  ""height"": $height$,
-  ""tiles"": [
-    { ""width"": 256, ""height"": 256, ""scaleFactors"": [ $scaleFactors$ ] }
-  ],
-  ""sizes"": [
-    $sizes$
-  ]
-}
-";
-            return InfoJson(serviceEndpoint, sizes, template, width, height);
+            var imageService = new ImageService2
+            {
+                Context = ImageService2.Image2Context,
+                Id = serviceEndpoint,
+                Type = null,
+                Protocol = ImageService2.Image2Protocol,
+                Profile = ImageService2.Level1Profile,
+                ProfileDescription = new ProfileDescription
+                {
+                    Formats = new[] { "jpg" },
+                    Qualities = new[] { "native", "color", "gray" },
+                    Supports = new[] { "regionByPct","sizeByForcedWh","sizeByWh","sizeAboveFull","rotationBy90s","mirroring","gray" }
+                }
+            };
+            SetSizes(imageService, sizes, width, height);
+            SetScaleFactors(imageService);
+            return imageService;
         }
         
         public static string GetImageApi2_1Level1Auth(string serviceEndpoint, int width, int height, List<int[]> sizes, string services)
```

Comment:
Eventually here I think we should ask Cantaloupe what info.json it would produce for this image, and then store the relevant variable bits of info. We'd still use the IIIF lib because we'd want to decorate the info.json with extra Orchestrator-level stuff like Auth. The reason for this is the tile size and scaleFactors - we assume always 256-size tiles and I don't want to assume that. But for now this is nice.

Reply:
At some point it might be better to make make this class non-static to ease the transition to Cantaloupe.